### PR TITLE
Fix timer

### PR DESCRIPTION
**Pull request by dr-orlovsky**

### CODE REVIEW

File: src/reactor.rs [hunk 1]
```diff
@@ -529,11 +529,16 @@ impl<H: Handler, P: Poll> Runtime<H, P> {
             // Blocking
             #[cfg(feature = "log")]
             log::trace!(target: "reactor", "Polling with timeout {timeout:?}");
-            match self.poller.poll(Some(timeout)) {
+
+            let res = self.poller.poll(Some(timeout));
+            let now =
+                SystemTime::now().duration_since(SystemTime::UNIX_EPOCH).expect("system time");
+
```
[the remark this answers is not in the document]
Comment (dr-orlovsky) [reply src/reactor.rs:537]: Done

**Review by cloudhead — APPROVED** [review]
LGTM

**Comment by cloudhead** [comment]
I can try testing it before you merge, if you want.

**Comment by cloudhead** [comment]
Ok it's still not working correctly. My hunch is that the timeout you're passing to `poll` is not correct, because the `Duration` type is being used for both absolute and relative time. When you get a timeout out of `TimeoutManager::next`, it's a unix time (absolute), but the `poll()` function expects a relative time to wait. Instead, it's getting a huge duration. So I guess you need to substract `before_poll`.

**Review by cloudhead — CHANGES_REQUESTED** [review]
See latest comment!

**Comment by dr-orlovsky** [comment]
@cloudhead did a quickfix

**Comment by dr-orlovsky** [comment]
Merged & released `v0.1.1`

**Pull request merged**